java-checks/src/main/java/org/sonar/java/checks/CallToDeprecatedMethodCheck.java

```diff
@@ -56,4 +74,26 @@ private boolean nonAbstractOrFlaggedForRemoval(Symbol.MethodSymbol method) {
       // if the method is flagged for removal, it will be handled by S5738
       || isFlaggedForRemoval(method));
   }
+
+  /**
+   * Tests that the arguments types match the parameter types.
+   * Please note that the method returns false when the signature contains variadic parameters.
+   *
+   * @param arguments Arguments in a method call
+   * @param types Parameter types in a method signature
+   * @return true if the arguments' types match types. false otherwise.
+   */
+  private static boolean argumentsMatchSignature(Arguments arguments, List<Type> types) {
+    if (arguments.size() != types.size()) {
+      return false;
+    }
+    for (int i = 0; i < arguments.size(); i++) {
+      ExpressionTree argument = arguments.get(i);
+      Type type = types.get(i);
+      if (!argument.symbolType().equals(type)) {
```

Comment:
It seems to me that this will be true only when `argument.symbolType()` is actually unknown. If this is correct, we can reduce the whole logic to something like: `return arguments.stream().noneMatch(a -> a.symbolType().isUnknown());`.
In addition, it will remove the FN with varargs.

What do you think about this solution?

Reply:
Good point, checking for "unknowness" seems to do the trick. Updating the PR now